cms/djangoapps/contentstore/views/preview.py

```diff
@@ -194,6 +196,10 @@ def _preview_module_system(request, descriptor, field_data):
         # stick the license wrapper in front
         wrappers.insert(0, partial(wrap_with_license, mako_service=mako_service))
 
+    preview_anonymous_user_id = 'student'
+    if individualize_anonymous_user_id(course_id):
+        preview_anonymous_user_id = anonymous_id_for_user(request.user, course_id)
```

Comment:
@schenedx  In order to ensure the same anonymous ID is used in Studio as in the LMS, you need to check the block descriptor to find out whether the anonymous user ID should be course-specific or not.

```suggestion
        if getattr(descriptor, 'requires_per_student_anonymous_id', False):
            preview_anonymous_user_id = anonymous_id_for_user(request.user, None)
        else:
            preview_anonymous_user_id = anonymous_id_for_user(request.user, course_id)
```

Reply:
I see. Thank you for pointing this out! This just points out my lack of knowledge in this area.
Made the change and please review again @pomegranited

Reply:
Thanks for making this change @schenedx ! I'm launching a sandbox but it's the end of my week -- I will test it out thoroughly on Monday and leave a review.

Reply:
Thanks @schenedx  -- I can review the code, but need a functional LTI provider in order to test it. And since Open edX no longer supports this with the new Learner MFE, I need your help with this step.

Can you provide a secret:key and URL that I can use to test this from my devstack?

Reply:
@pomegranited You can use https://saltire.lti.app/tool (check this URL for config parameters, but the values below should work).

- Launch URL: https://saltire.lti.app/tool
- ID: `test`
- LTI passport: `test:jisc.ac.uk:secret`

Reply:
You're the best @giovannicimolin !